### PR TITLE
*: remove `IsFeatureSupported` from `Cluster` interface

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Ref #4340. 

### What is changed and how it works?

Just as the title says.

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/mock/mockcluster/mockcluster.go [hunk 1]
```diff
@@ -729,26 +729,6 @@ func (mc *Cluster) SetStoreLabel(storeID uint64, labels map[string]string) {
 	mc.PutStore(newStore)
 }
 
-// DisableFeature marks that these features are not supported in the cluster.
-func (mc *Cluster) DisableFeature(fs ...versioninfo.Feature) {
-	for _, f := range fs {
-		mc.disabledFeatures[f] = struct{}{}
```
Comment (HunDunDM) [inline pkg/mock/mockcluster/mockcluster.go:735]: Also delete `disabledFeatures`?

File: server/schedule/operator/builder_test.go [hunk 2]
```diff
@@ -141,6 +143,7 @@ func (s *testBuilderSuite) TestPrepareBuild(c *C) {
 	c.Assert(builder.currentLeaderStoreID, Equals, uint64(1))
 
 	// do not use joint consensus
+	s.cluster.SetClusterVersion(versioninfo.MinSupportedVersion(versioninfo.Version4_0))
```
Comment (HunDunDM) [inline server/schedule/operator/builder_test.go:146]: This setting is not dependent here.

File: server/schedule/operator/builder_test.go [hunk 3]
```diff
@@ -119,6 +120,7 @@ func (s *testBuilderSuite) TestPrepareBuild(c *C) {
 	c.Assert(err, NotNil)
 
 	// use joint consensus
+	s.cluster.SetClusterVersion(versioninfo.MinSupportedVersion(versioninfo.JointConsensus))
```
Comment (HunDunDM) [inline server/schedule/operator/builder_test.go:123]: It should be an initial value.

File: server/schedule/operator/create_operator_test.go [hunk 4]
```diff
@@ -159,6 +159,8 @@ func (s *testCreateOperatorSuite) TestCreateSplitRegionOperator(c *C) {
 }
 
 func (s *testCreateOperatorSuite) TestCreateMergeRegionOperator(c *C) {
+	// use joint consensus
+	s.cluster.SetClusterVersion(versioninfo.MinSupportedVersion(versioninfo.JointConsensus))
```
Comment (HunDunDM) [inline server/schedule/operator/create_operator_test.go:163]: ditto

File: server/schedule/operator/create_operator_test.go [hunk 5]
```diff
@@ -564,6 +566,8 @@ func (s *testCreateOperatorSuite) TestCreateLeaveJointStateOperator(c *C) {
 }
 
 func (s *testCreateOperatorSuite) TestCreateMoveRegionOperator(c *C) {
+	// use joint consensus
+	s.cluster.SetClusterVersion(versioninfo.MinSupportedVersion(versioninfo.JointConsensus))
```
Comment (HunDunDM) [inline server/schedule/operator/create_operator_test.go:570]: ditto

**Review by HunDunDM — COMMENTED** [review]
The `mockcluster` should get the initial value of the version at `NewCluster`, which is the maximum value in `versioninfo`.

**Review by lhy1024 — APPROVED** [review]

**Comment by rleungx** [comment]
@HunDunDM PTAL

**Review by nolouch — APPROVED** [review]
lgtm

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**